csrc/python_frontend/fusion_record.h

```diff
@@ -1264,6 +1278,17 @@ struct TensorRecord : RecordFunctor {
       NVF_CHECK(!is_cpu_, "CPU non-scalar tensor is not supported!");
     }
 
+    if (!stride_order_.empty()) {
+      std::vector<IterDomain*> allocation_domain(rank);
+      for (auto i : c10::irange(rank)) {
+        allocation_domain[rank - 1 - static_cast<int>(stride_order_[i])] =
```

Comment:
Is there any invariant we need to enforce on `stride_order`?
`NVF_ERROR(stride_order > 0 && stride_order < rank, "stride order must be between 0 and tensor rank.")`

Reply:
we can support negative indexing. I'll add that.